tests/unit/optimizers/test_lr_scheduler.py

```diff
@@ -0,0 +1,434 @@
+import torch
+import torch.nn.functional as F
+from torch.optim import SGD
+from torch.optim.lr_scheduler import _LRScheduler
+
+import math
+import numpy as np
+from tests import reset_seed
+import pytorch_lightning as pl
+from pl_bolts.optimizers.lr_scheduler import LinearWarmupCosineAnnealingLR
+
+
+EPSILON = 1e-12
```

Comment:
I think there is an epsilon defined somewhere in lightning iirc, maybe use this to avoid duplication?

cc @Borda who might know, where it is ^^

Reply:
@Borda so I had this question about default epsilons in lightning, ```numpy.finfo(numpy.float32).eps``` is 1.1920929e-07. Since learning rates used are up to 1e-6, we need a higher precision in schedulers. But, ```numpy.finfo(numpy.float64).eps``` comes out to be 2.220446049250313e-16, which results in failing tests. So the ideal epsilon seemed 1e-12.